pss/crypto.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+package pss
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	ethCrypto "github.com/ethereum/go-ethereum/crypto"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv6"
+)
+
+var cryptoBackend defaultCryptoBackend
```

Comment:
Perhaps we should not use a singleton. Can we imagine different topics using different backends?

Reply:
Does it make sense that at this stage? The purpose of this PR is to isolate the currrent whisper+crypto code in order to do further improvements. Now, the crypto provider is singleton (because we only user whisper/ethereum crypto). 
Wouldn't it be better to do that kind of things in further PR's?

Reply:
Ok let's do that